src/core/auth.ts

```diff
@@ -50,3 +65,27 @@ export function composeAuthMsg(
   msg += `Expiration Time: ${authParam.expiration_time}\n`;
   return msg;
 }
+
+export function removeTrailingSlash(url: string): string {
+  if (url.endsWith('/')) {
+    return url.slice(0, -1);
+  }
+  return url;
+}
+
+export function verifyAuthProperties(
+  authParm: AuthInfo,
+  domain: string,
+  version: string,
+  chainId: string,
+): void {
+  if (authParm.domain !== domain) {
```

Comment:
I think this is problematic if run against older kgw version?  Since old kgw version won't return those fields.

Reply:
That is a good point - how are you handling this with Kwil-cli? Maybe we update the version number? So if > version 1, then run the `verifyAuthProperties`?

Reply:
We didn't change 'version', since 'version' is for something like template.

https://github.com/kwilteam/kwil-db/pull/657/files#diff-2a7eaeea47cbb7c8cd1df3a3972976df3131d7634f9d6a1d26d2409ee064718cR165  I only compare if the field is returned from KGW